orangecontrib/bioinformatics/tests/annotation/test_annotate_samples.py

```diff
@@ -242,3 +242,49 @@ def test_entrez_id_not_string(self):
         self.assertGreater(np.nansum(annotations.X), 0)
         self.assertLessEqual(np.nanmax(annotations.X), 1)
         self.assertGreaterEqual(np.nanmin(annotations.X), 0)
+
+    def test_celltypes_as_feature(self):
+        """
+        Test case when cell type or genes appears as a feature
+        """
+        ct_values = self.markers.get_column_view("Cell Type")[0].tolist()
+        entrezid = self.markers.get_column_view("Entrez ID")[0].tolist()
+        ct_variable = DiscreteVariable("Cell Type", values=list(set(ct_values)))
+        entrez_variable = DiscreteVariable("Entrez ID", values=list(set(entrezid)))
+
+        markers = Table(
+            Domain([ct_variable, entrez_variable]),
+            list(
+                zip(
+                    [ct_variable.values.index(x) for x in ct_values],
+                    [entrez_variable.values.index(x) for x in entrezid],
+                )
+            ),
+        )
+        annotations = self.annotate_samples(self.data, markers)
+
+        self.assertEqual(type(annotations), Table)
+        self.assertEqual(len(annotations), len(self.data))
+        self.assertEqual(len(annotations[0]), 2)  # two types in the data
+        self.assertGreater(np.nansum(annotations.X), 0)
+        self.assertLessEqual(np.nanmax(annotations.X), 1)
+        self.assertGreaterEqual(np.nanmin(annotations.X), 0)
+
+        # one in metas one in features
+        markers = Table(
```

Comment:
This seems like it would belong to a separate test. You're testing two different things here and the test is long.

Reply:
It is fixed now